service/history/execution/mutable_state_builder.go

```diff
@@ -4811,24 +4812,30 @@ func (e *mutableStateBuilder) unixNanoToTime(
 }
 
 func (e *mutableStateBuilder) logInfo(msg string, tags ...tag.Tag) {
-	tags = append(tags, tag.WorkflowID(e.executionInfo.WorkflowID))
-	tags = append(tags, tag.WorkflowRunID(e.executionInfo.RunID))
-	tags = append(tags, tag.WorkflowDomainID(e.executionInfo.DomainID))
-	e.logger.Info(msg, tags...)
+	if e != nil && e.executionInfo != nil {
```

Comment:
do we need these nil checks? If so let's check for nilness and terminate early.
```
if e == nil || e.executionInfo == nil {
  return
}

// log with tags
```

Reply:
ah, sorry about that, let me address in a follow up

Reply:
https://github.com/uber/cadence/pull/5776